lib/javascript/JavascriptModulesPlugin.js

```diff
@@ -850,7 +850,7 @@ class JavascriptModulesPlugin {
 			runtimeRequirements.has(RuntimeGlobals.moduleFactoriesAddOnly) ||
 			runtimeRequirements.has(RuntimeGlobals.require)
 		) {
-			source.add(`${prefix}var __webpack_modules__ = (`);
+			source.add(`${prefix}var ${RuntimeGlobals.webpackModules} = (`);
```

Comment:
Hm, maybe I was wrong, it can be misleading, because here `__webpack__modules__` is cache for loading modules, for ESM `__webpack__modules__` is place where we store new modules, maybe we should use another name for ESM runtime variables, for example - `RuntimeGlobals.ESMWebpackModules`, and do not replace here to avoid misleading... What do you think?

Reply:
@alexander-akait  I think it should be like this. I'll make the changes.